components/hab/src/command/origin/key/download.rs

```diff
@@ -187,7 +184,7 @@ fn download_key(ui: &mut UI,
                 Ok(())
             };
 
-            if retry(wait_for(RETRY_WAIT, RETRIES), download_fn).is_err() {
+            if retry(delay::Fixed::from(RETRY_WAIT).take(RETRIES), download_fn).is_err() {
```

Comment:
Would it be worth converting this to use `map_err` along with `?` and get rid of the explicit return?

```rust
retry(...).map_err(|_| ...)?;
```